src/controller/cna.controller/cna.controller.js

```diff
@@ -34,232 +34,232 @@ async function getCna (req, res) {
-    if (typeof req.body.name !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
-
-    cna.name = req.body.name
+  if (cna.short_name === undefined) {
+    cna.short_name = shortName
   }
 
+  const newCna = new Cna(cna)
   Cna.findOneAndUpdate()
     .byShortName(shortName)
-    .updateOne(cna)
+    .updateOne(newCna)
     .exec((err, result) => {
       if (err) {
         logger.error(err.stack)
         return res.status(500).send('Internal Server Error')
       }
 
       if (result.n === 0) {
-        logger.info(shortName + ' CNA was not updated in MongoDB because it does not exists in the database.')
-        return res.status(200).json({ message: 'The cna does not exist' })
-      } else {
-        logger.info(shortName + ' CNA was successfully updated in MongoDB')
+        logger.info(shortName + ' CNA could not be updated in MongoDB because it does not exist.')
+        return res.status(404).json({ message: shortName + ' CNA does not exist.' })
       }
 
       const responseMessage = {
-        message: 'The cna was successfully updated',
-        updated: cna
+        message: 'The cna was successfully updated.',
+        updated: newCna
       }
 
+      logger.info(JSON.stringify(responseMessage))
       return res.status(200).json(responseMessage)
     })
 }
 
-// TODO: Request body sanitation and parameter validation
-// Updates a user if one exist for the specified username. If one doesn't exists, it creates a user.
-async function updateUser (req, res) {
-  const username = req.params.id
-  const user = {}
-
-  if (req.body.username === undefined) {
-    return res.status(400).json({ message: 'Provide the user\'s username to create or update the user' })
-  } else {
-    if (typeof req.body.username !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
-
-    user.username = req.body.username
-  }
-
-  if (req.body.cna_name !== undefined) {
-    if (typeof req.body.cna_name !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
+// Creates a user only if the cna exist and the user doesn't exist for the specified shortname and username respectively.
+async function createUser (req, res) {
+  const cnaShortName = req.params.shortname
+  const user = req.body
 
-    user.cna_name = req.body.cna_name
+  if (user.username === undefined) {
+    return res.status(400).json({ message: 'To create a user, a username must be provided.' })
   }
 
-  if (req.body.cna_short_name !== undefined) {
-    if (typeof req.body.cna_short_name !== 'string') {
-      return res.status(400).json({ message: 'Wrong value type' })
-    }
-
-    user.cna_short_name = req.body.cna_short_name
+  if (user.cna_short_name !== undefined && user.cna_short_name !== cnaShortName) {
+    return res.status(400).json({ message: cnaShortName + ' URL parameter does not match ' + user.cna_short_name + ' in the JSON body.' })
   }
```

Comment:
Include the following else statement to use the cnaShortName in case the user didn't provide it in the request.
} else {
    newUser.cna_short_name = cnaShortName
  }

Reply:
Instead of an else, I thought an if with the clear condition was more readable. So I put that above this logic.